llvm/test/Transforms/JumpThreading/preserving-debugloc-fold-select.ll

```diff
@@ -0,0 +1,98 @@
+; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 4
```

Comment:
The checks in this test are generated by using update_test_checkes.py, and then I remove some redundant debug metadata checks and keep the `DBG29`. So, if someone regenerate the checks, the crucial checks would still be there.

For those instruction checks, I think we can only keep those of `.exit.thread4`.

Reply:
You should not modify or remove checks that were automatically generated. That's the point of automatic generation.

The automated scripts generate far more checks than are actually necessary, but removing some doesn't actually help anything.

Reply:
> You should not modify or remove checks that were automatically generated. That's the point of automatic generation.

So, should I keep all the automatically generated checks? How about those redundant checks that are irrelevant to the fix in this case?

Reply:
I think removing the line `; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 4` here should be enough for this one (discussion continues up in the other test though).

Otherwise the latest change to this one looks good, thanks.

Reply:
Right. If the test has a comment saying the tests were auto-generated, that comment needs to tell the truth (no manual modifications of the checks). It's likely the generated CHECKs will do more than you would in a manual test, but that's okay. People using the scripts expect that.

Reply:
Oh, I understand now! Thanks for pointing out this!